pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/TableRebalancer.java

```diff
@@ -300,4 +399,55 @@ private void waitForStable(String resourceName)
       }
     } while (diff > 0);
   }
+
+  /**
+   * Helper class that maintains stats that
+   * are later checked in tests to verify
+   * the behavior of the algorithm here
+   * that takes from current ideal state
+   * to a target ideal state
+   */
+  public static class RebalancerStats {
+    private int dryRun;
+    private int directTransitions;
+    private int incrementalTransitions;
+
+    RebalancerStats() {
+    }
+
+    /**
+     * Number of dry runs. Can only be 1
+     * @return
+     */
+    public int getDryRun() {
+      return dryRun;
+    }
+
+    /**
+     * Get the number of times we updated the ideal
+     * state at one go. This happens in downtime
+     * rebalancing and can also happen in no-downtime
+     * rebalancing if there are sufficient common
+     * hosts between current and target ideal states
+     * @return direct transitions
+     */
+    public int getDirectTransitions() {
+      return directTransitions;
+    }
+
+    /**
+     * Get the number of times we updated the ideal
```

Comment:
this is really number of segments we updated, because we bump this once for each segment that we change in idealstate. We may then update the idealstate in one go for all the segments.  We should also keep the number of idealstate updates as a separate stat.

Reply:
changed the name of these stats to avoid confusion, also tracking the number of times we updated the ideal state in ZK (which happens after we have made updates to instance map of each segment)